circleci: failed tests: TestStoreRemovePlaceholderOnRaftIgnored

The following test appears to have failed:

[#22136](https://circleci.com/gh/cockroachdb/cockroach/22136):

```
I160901 14:29:26.912690 5618 storage/replica.go:2776  gossiping first range from store 1, range 1: [{1 1 1}]
I160901 14:29:26.913945 5620 storage/replica.go:2769  [s1,r1{/Min-/Max}] gossiping sentinel from store 1, range 1
I160901 14:29:26.914270 5620 storage/replica.go:2776  [s1,r1{/Min-/Max}] gossiping first range from store 1, range 1: [{1 1 1}]
W160901 14:29:26.916724 5595 storage/replica.go:1936  [s1,r1{/Min-/Min}] failed to look up recipient replica 0 in range 1 while sending MsgAppResp: replica 0 not present in {2 2 2}, []
I160901 14:29:26.919889 5591 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- FAIL: TestStoreRemovePlaceholderOnRaftIgnored (0.04s)
	store_test.go:2386: expected 1 dropped placeholder, but found 0
=== RUN   TestStoresAddStore
--- PASS: TestStoresAddStore (0.01s)
=== RUN   TestStoresRemoveStore
--- PASS: TestStoresRemoveStore (0.02s)
=== RUN   TestStoresGetStoreCount
--- PASS: TestStoresGetStoreCount (0.04s)
=== RUN   TestStoresVisitStores
--- PASS: TestStoresVisitStores (0.00s)
=== RUN   TestStoresGetStore
--
=== RUN   Example_rebalancing
I160901 14:30:17.046090 1 gossip/gossip.go:230  initial resolvers: []
W160901 14:30:17.046192 1 gossip/gossip.go:1041  no resolvers found; use --join to specify a connected node
I160901 14:30:17.046428 1 gossip/gossip.go:264  NodeID set to 1
--- PASS: Example_rebalancing (1.29s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	71.682s
=== RUN   TestBatchBasics
I160901 14:30:33.121168 8 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160901 14:30:33.124623 8 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.98s)
=== RUN   TestBatchRepr
I160901 14:30:34.099505 9 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160901 14:30:34.103380 9 storage/engine/rocksdb.go:402  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.02s)
=== RUN   TestBatchGet
I160901 14:30:34.116675 24 storage/engine/rocksdb.go:354  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/store_test.go
+++ b/storage/store_test.go
@@ -2376,13 +2376,15 @@ func TestStoreRemovePlaceholderOnRaftIgnored(t *testing.T) {
 		numPlaceholders := len(s.mu.replicaPlaceholders)
 		s.mu.Unlock()
 
-		if numPlaceholders == 0 {
-			return nil
+		if numPlaceholders != 0 {
+			return errors.Errorf("expected 0 placeholders, but found %d", numPlaceholders)
 		}
-		return errors.Errorf("expected 0 placeholders, but found %d", numPlaceholders)
+		// The count of dropped placeholders is incremented after the placeholder
+		// is removed (and while not holding Store.mu), so we need to perform the
+		// check of the number of dropped placeholders in this retry loop.
+		if n := atomic.LoadInt32(&s.counts.droppedPlaceholders); n != 1 {
+			return errors.Errorf("expected 1 dropped placeholder, but found %d", n)
+		}
+		return nil
 	})
-
-	if n := atomic.LoadInt32(&s.counts.droppedPlaceholders); n != 1 {
-		t.Fatalf("expected 1 dropped placeholder, but found %d", n)
-	}
 }

